simd/src/Kokkos_SIMD_NEON.hpp

```diff
@@ -943,6 +1130,66 @@ class where_expression<simd_mask<double, simd_abi::neon_fixed_size<2>>,
   }
 };
 
+template <>
+class const_where_expression<simd_mask<float, simd_abi::neon_fixed_size<2>>,
+                             simd<float, simd_abi::neon_fixed_size<2>>> {
+ public:
+  using abi_type   = simd_abi::neon_fixed_size<2>;
+  using value_type = simd<float, abi_type>;
+  using mask_type  = simd_mask<float, abi_type>;
+
+ protected:
+  value_type& m_value;
+  mask_type const& m_mask;
+
+ public:
+  const_where_expression(mask_type const& mask_arg, value_type const& value_arg)
+      : m_value(const_cast<value_type&>(value_arg)), m_mask(mask_arg) {}
+
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+  void copy_to(float* mem, element_aligned_tag) const {
+    if (m_mask[0]) mem[0] = m_value[0];
+    if (m_mask[1]) mem[1] = m_value[1];
+  }
```

Comment:
`scatter_to`?

Reply:
Added in [0508a93](https://github.com/kokkos/kokkos/pull/6177/commits/0508a9321685d1feabdf586c2c2fbe1d9e68eba4).